bus-mapping/src/evm/opcodes.rs

```diff
@@ -324,10 +324,11 @@ pub fn gen_begin_tx_ops(state: &mut CircuitInputStateRef) -> Result<(), Error> {
         },
     )?;
 
+    #[allow(clippy::if_same_then_else)]
     if call.is_create() {
-        unimplemented!("Creation transaction is not yet implemented")
+        // TODO: Implement creation transaction
     } else if state.is_precompiled(&call.address) {
-        unimplemented!("Call to precompiled is not yet implemented")
+        // TODO: Implement calling to precompiled
     } else {
```

Comment:
A `match` could serve better here maybe.
Aside from that, no issues.

Reply:
You mean something like this, right?

```rust
match (call.is_create(), state.is_precompiled(&call.address)) {
    (true, _) => {
        // TODO: Implement creation transaction
    }
    (_, true) => {
        // TODO: Implement calling to precompiled
    }
    _ => {
        state.push_op(
            RW::READ,
            AccountOp {
                address: call.address,
                field: AccountField::CodeHash,
                value: code_hash.to_word(),
                value_prev: code_hash.to_word(),
            },
        );
    }
}
```

Reply:
Yup. Generaly, if/else is recomended only for true/false variants.

If more variants are added (else_if) `match` is usually the recomendation. As it allows for easier upgrades in the workflow.

Not a requirement for this PR if you prefer to stick to if else statements.

Reply:
Fixed in [`5d2c6a6`](https://github.com/appliedzkp/zkevm-circuits/pull/389/commits/5d2c6a6a0ddb50f982b289b48cb9527db3501454). Thanks!